config/environments/production.rb

```diff
@@ -109,4 +110,17 @@
   # config.active_record.database_selector = { delay: 2.seconds }
   # config.active_record.database_resolver = ActiveRecord::Middleware::DatabaseSelector::Resolver
   # config.active_record.database_resolver_context = ActiveRecord::Middleware::DatabaseSelector::Resolver::Session
+  #
+  config.action_mailer.perform_deliveries = true
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.default_url_options = { :host => "https://fathomless-earth-16928.herokuapp.com" }
```

Comment:
Was passiert, wenn man die App auf Heroku umbenennt oder auf einem AWS-Server hosten will?
Es ist nie gut, solche Variablen fest im Code einzutragen. Besser ist es diese Richtlinien zu folgen: https://12factor.net/

Beispiel ist in der Zeile 121 zu sehen. Dazu einfach die Heroku Doku aufmachen https://devcenter.heroku.com/articles/sendgrid#actionmailer.

Reply:
@viktorianer  ICh habe das jetzt geändert, also die host url in eine config var auf Heroku gespeichert und nutze diese bei default_url_options.

Soweit es aus den Heroku logs hervor geht funktioniert es jetzt.

Allerdings wird mein SendGrid Account jedesmal direkt bei der ersten Mail Geblockt/ Gebannt
(Hab das jetzt zweimal mit jeweils einer neuen Heroku App versucht)

Ich bekomme keine Email von Heroku oder SendGrid in welcher steht woran es liegt, ich kann danach auch nicht mehr auf das SendGrid Dashboard der Heroku App zugreifen.

Ich habe ansonsten eigentlich nichts am code verändert und auch keinerlei Idee woran das jetzt liegt.

Siehst du irgendetwas im Code  oder hast eine Idee / Erfahrungen mit Sendgrid in der Beziehung?

Bin gerade ein wenig Ratlos

Reply:
Für Sendgrid wird eine Kreditkarte benötigt. Das kannst du in der Doku von Heroku finden. Es wird auch kein Geld abgebucht, weil es kostenlos ist.
Für uns reicht es, dass es funktioniert. Also, kannst du weiter machen.

Reply:
Das habe ich schon gemacht.

Hatte es ja schon mal alles am Laufen als ich die Pull Request geöffnet habe

Reply:
Naja, die Konfiguration ist noch nicht ganz richtig. Ist an sich egal jetzt, es ging mir darum, dass du das Grundprinzip von https://12factor.net/ kennst und anwendest.

Reply:
Hmm funktioniert aber trotzdem nicht.
Sobald die erste mail versendet werden soll wird meine SendGrid geblockt.

![1](https://user-images.githubusercontent.com/44459242/74246489-e2d3b080-4ce4-11ea-9f6c-24aaaeb82550.jpg)
![2](https://user-images.githubusercontent.com/44459242/74246502-e830fb00-4ce4-11ea-8232-1a24cd121efd.jpg)
![3](https://user-images.githubusercontent.com/44459242/74246506-ea935500-4ce4-11ea-8238-fad259a4a8fb.jpg)

Reply:
Du muss folgendes in deiner App machen:
`heroku addons:create sendgrid:starter`
Dann die `config/environments/production.rb` anpassen, was du bereits gemacht hast.

Prüfe dann, ob deine ENV richtig geliefert werden:
```
heroku config:get SENDGRID_USERNAME
heroku config:get SENDGRID_PASSWORD
```

Dann sollte es gehen. Wenn nicht, finde ich den Fehler nicht so einfach. Es sollte aber sofort und ohne Umwege klappen, wenn man es wie in der Doku beschrieben ist, macht. :)

Reply:
Okay habs ...keine Ahnung woran es lag .
Habe das jetzt alles nochmal in ein Altes Heroku Repo gepusht.
Da hats dann funktioniert.

Danke aber für die schnellen antworten